app/controllers/BtaStubController.scala

```diff
@@ -22,16 +22,23 @@ import config.AppConfig
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent}
 import services.BtaStubService
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions, NoActiveSession}
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
+import scala.concurrent.Future
+
 @Singleton
-class BtaStubController @Inject()(val messagesApi: MessagesApi,
-                                  btaStubService: BtaStubService,
-                                  implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
+class BtaStubController @Inject()(val messagesApi: MessagesApi, val authFunctions: AuthorisedFunctions,
+                                  val authConnector: AuthConnector, btaStubService: BtaStubService, implicit val appConfig: AppConfig)
```

Comment:
There is an `EnrolmentsAuthService` already that extends `AuthorisedFunctions` that you can inject instead and call `.authorised()` on